File: app/src/main/java/com/ash/studios/musify/Models/Playlist.java
package com.ash.studios.musify.Models;

import androidx.annotation.NonNull;

import java.io.Serializable;
import java.util.ArrayList;
import java.util.Objects;

public class Playlist implements Serializable {
    private final String name;
    private final ArrayList<Song> songs;

    public Playlist(String name, ArrayList<Song> songs) {
        this.name = name;
        this.songs = songs;
    }

    public String getName() {
        return name;
    }

    public ArrayList<Song> getSongs() {
        return songs;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof Playlist)) return false;
        Playlist playlist = (Playlist) o;
        return Objects.equals(getName(), playlist.getName()) &&
                Objects.equals(getSongs(), playlist.getSongs());
    }

    @Override
    public int hashCode() {
        return Objects.hash(getName(), getSongs());
    }

    @NonNull
    @Override
    public String toString() {
        return "Playlist{" +
                "name='" + name + '\'' +
                ", songs=" + songs +
                '}';
    }
}

File: app/src/main/java/com/ash/studios/musify/models/Playlist.kt
package com.ash.studios.musify.models

import java.io.Serializable
import java.util.*

class Playlist(val name: String, val songs: ArrayList<Song>) : Serializable {

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other !is Playlist) return false
        return name == other.name && songs == other.songs
    }

    override fun hashCode(): Int {
        return Objects.hash(name, songs)
    }

    override fun toString(): String {
        return "Playlist{" +
                "name='" + name + '\'' +
                ", songs=" + songs +
                '}'
    }
}